### ISSUE TITLE
Can't build Client with mingw32-make.exe on Windows 10

### ISSUE THREAD

**Issue by jmoseman01** [Rinnegatamante/VitaPad#19]

**Comment by jmoseman01** [comment]
C:\Users\jmose\source\repos\VitaPad\Client>mingw32-make.exe
g++ -c \
        -static -static-libgcc -static-libstdc++ \
        -I ViGEmClient/include \
        -municode -DERROR_INVALID_DEVICE_OBJECT_PARAMETER=650L\
        ViGEmClient/src/ViGEmClient.cpp
In file included from ViGEmClient/src/ViGEmClient.cpp:54:
ViGEmClient/src/Internal.h:72:35: error: 'thread' is not a member of 'std'
  std::unique_ptr<std::vector<std::thread>> notificationThreadList;
                                   ^~~~~~
ViGEmClient/src/Internal.h:72:35: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/Internal.h:1:1:
+#include <thread>
 /*
ViGEmClient/src/Internal.h:72:35:
  std::unique_ptr<std::vector<std::thread>> notificationThreadList;
                                   ^~~~~~
ViGEmClient/src/Internal.h:72:35: error: 'thread' is not a member of 'std'
ViGEmClient/src/Internal.h:72:35: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/Internal.h:72:35: error: template argument 1 is invalid
ViGEmClient/src/Internal.h:72:35: error: template argument 2 is invalid
ViGEmClient/src/Internal.h:72:41: error: template argument 1 is invalid
  std::unique_ptr<std::vector<std::thread>> notificationThreadList;
                                         ^~
ViGEmClient/src/Internal.h:72:41: error: template argument 2 is invalid
ViGEmClient/src/ViGEmClient.cpp: In function '_VIGEM_TARGET_T* VIGEM_TARGET_ALLOC_INIT(VIGEM_TARGET_TYPE)':
ViGEmClient/src/ViGEmClient.cpp:157:35: error: cannot convert 'std::nullptr_t' to 'int' in assignment
  target->notificationThreadList = nullptr;
                                   ^~~~~~~
ViGEmClient/src/ViGEmClient.cpp: In function 'VIGEM_ERROR vigem_target_add_async(PVIGEM_CLIENT, PVIGEM_TARGET, PFN_VIGEM_TARGET_ADD_RESULT)':
ViGEmClient/src/ViGEmClient.cpp:457:10: error: 'thread' is not a member of 'std'
     std::thread _async{ [](
          ^~~~~~
ViGEmClient/src/ViGEmClient.cpp:457:10: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/ViGEmClient.cpp:55:1:
+#include <thread>

ViGEmClient/src/ViGEmClient.cpp:457:10:
     std::thread _async{ [](
          ^~~~~~
ViGEmClient/src/ViGEmClient.cpp:504:5: error: '_async' was not declared in this scope
     _async.detach();
     ^~~~~~
ViGEmClient/src/ViGEmClient.cpp: In function 'VIGEM_ERROR vigem_target_x360_register_notification(PVIGEM_CLIENT, PVIGEM_TARGET, PFN_VIGEM_X360_NOTIFICATION, LPVOID)':
ViGEmClient/src/ViGEmClient.cpp:690:36: error: invalid operands of types 'int' and 'std::nullptr_t' to binary 'operator=='
  if(target->notificationThreadList == nullptr)
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
ViGEmClient/src/ViGEmClient.cpp:691:70: error: 'thread' is not a member of 'std'
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                      ^~~~~~
ViGEmClient/src/ViGEmClient.cpp:691:70: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/ViGEmClient.cpp:691:70: error: 'thread' is not a member of 'std'
ViGEmClient/src/ViGEmClient.cpp:691:70: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/ViGEmClient.cpp:691:70: error: template argument 1 is invalid
ViGEmClient/src/ViGEmClient.cpp:691:70: error: template argument 2 is invalid
ViGEmClient/src/ViGEmClient.cpp:691:79: error: no matching function for call to 'make_unique<<expression error> >()'
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                               ^
In file included from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/memory:80,
                 from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/thread:39,
                 from ViGEmClient/src/ViGEmClient.cpp:48:
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:830:5: note: candidate: 'template<class _Tp, class ... _Args> typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...)'
     make_unique(_Args&&... __args)
     ^~~~~~~~~~~
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:830:5: note:   template argument deduction/substitution failed:
ViGEmClient/src/ViGEmClient.cpp:691:79: error: template argument 1 is invalid
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                               ^
In file included from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/memory:80,
                 from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/thread:39,
                 from ViGEmClient/src/ViGEmClient.cpp:48:
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:836:5: note: candidate: 'template<class _Tp> typename std::_MakeUniq<_Tp>::__array std::make_unique(std::size_t)'
     make_unique(size_t __num)
     ^~~~~~~~~~~
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:836:5: note:   template argument deduction/substitution failed:
ViGEmClient/src/ViGEmClient.cpp:691:79: error: template argument 1 is invalid
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                               ^
In file included from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/memory:80,
                 from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/thread:39,
                 from ViGEmClient/src/ViGEmClient.cpp:48:
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:842:5: note: candidate: 'template<class _Tp, class ... _Args> typename std::_MakeUniq<_Tp>::__invalid_type std::make_unique(_Args&& ...)' <deleted>
     make_unique(_Args&&...) = delete;
     ^~~~~~~~~~~
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:842:5: note:   template argument deduction/substitution failed:
ViGEmClient/src/ViGEmClient.cpp:691:79: error: template argument 1 is invalid
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                               ^
ViGEmClient/src/ViGEmClient.cpp:703:32: error: base operand of '->' is not a pointer
  target->notificationThreadList->emplace_back(std::thread(&vigem_notification_thread_worker, vigem, target, std::move(payloadVector)));
                                ^~
ViGEmClient/src/ViGEmClient.cpp:703:52: error: 'thread' is not a member of 'std'
  target->notificationThreadList->emplace_back(std::thread(&vigem_notification_thread_worker, vigem, target, std::move(payloadVector)));
                                                    ^~~~~~
ViGEmClient/src/ViGEmClient.cpp:703:52: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/ViGEmClient.cpp: In function 'VIGEM_ERROR vigem_target_ds4_register_notification(PVIGEM_CLIENT, PVIGEM_TARGET, PFN_VIGEM_DS4_NOTIFICATION, LPVOID)':
ViGEmClient/src/ViGEmClient.cpp:738:37: error: invalid operands of types 'int' and 'std::nullptr_t' to binary 'operator=='
  if (target->notificationThreadList == nullptr)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
ViGEmClient/src/ViGEmClient.cpp:739:70: error: 'thread' is not a member of 'std'
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                      ^~~~~~
ViGEmClient/src/ViGEmClient.cpp:739:70: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/ViGEmClient.cpp:739:70: error: 'thread' is not a member of 'std'
ViGEmClient/src/ViGEmClient.cpp:739:70: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/ViGEmClient.cpp:739:70: error: template argument 1 is invalid
ViGEmClient/src/ViGEmClient.cpp:739:70: error: template argument 2 is invalid
ViGEmClient/src/ViGEmClient.cpp:739:79: error: no matching function for call to 'make_unique<<expression error> >()'
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                               ^
In file included from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/memory:80,
                 from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/thread:39,
                 from ViGEmClient/src/ViGEmClient.cpp:48:
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:830:5: note: candidate: 'template<class _Tp, class ... _Args> typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...)'
     make_unique(_Args&&... __args)
     ^~~~~~~~~~~
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:830:5: note:   template argument deduction/substitution failed:
ViGEmClient/src/ViGEmClient.cpp:739:79: error: template argument 1 is invalid
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                               ^
In file included from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/memory:80,
                 from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/thread:39,
                 from ViGEmClient/src/ViGEmClient.cpp:48:
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:836:5: note: candidate: 'template<class _Tp> typename std::_MakeUniq<_Tp>::__array std::make_unique(std::size_t)'
     make_unique(size_t __num)
     ^~~~~~~~~~~
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:836:5: note:   template argument deduction/substitution failed:
ViGEmClient/src/ViGEmClient.cpp:739:79: error: template argument 1 is invalid
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                               ^
In file included from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/memory:80,
                 from C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/thread:39,
                 from ViGEmClient/src/ViGEmClient.cpp:48:
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:842:5: note: candidate: 'template<class _Tp, class ... _Args> typename std::_MakeUniq<_Tp>::__invalid_type std::make_unique(_Args&& ...)' <deleted>
     make_unique(_Args&&...) = delete;
     ^~~~~~~~~~~
C:/Program Files/mingw-w64/x86_64-8.1.0-win32-seh-rt_v6-rev0/mingw64/lib/gcc/x86_64-w64-mingw32/8.1.0/include/c++/bits/unique_ptr.h:842:5: note:   template argument deduction/substitution failed:
ViGEmClient/src/ViGEmClient.cpp:739:79: error: template argument 1 is invalid
   target->notificationThreadList = std::make_unique<std::vector<std::thread>>();
                                                                               ^
ViGEmClient/src/ViGEmClient.cpp:749:32: error: base operand of '->' is not a pointer
  target->notificationThreadList->emplace_back(std::thread(&vigem_notification_thread_worker, vigem, target, std::move(payloadVector)));
                                ^~
ViGEmClient/src/ViGEmClient.cpp:749:52: error: 'thread' is not a member of 'std'
  target->notificationThreadList->emplace_back(std::thread(&vigem_notification_thread_worker, vigem, target, std::move(payloadVector)));
                                                    ^~~~~~
ViGEmClient/src/ViGEmClient.cpp:749:52: note: 'std::thread' is defined in header '<thread>'; did you forget to '#include <thread>'?
ViGEmClient/src/ViGEmClient.cpp: In function 'void vigem_target_x360_unregister_notification(PVIGEM_TARGET)':
ViGEmClient/src/ViGEmClient.cpp:761:37: error: invalid operands of types 'int' and 'std::nullptr_t' to binary 'operator!='
  if (target->notificationThreadList != nullptr)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
ViGEmClient/src/ViGEmClient.cpp:764:47: error: base operand of '->' is not a pointer
   std::for_each(target->notificationThreadList->begin(), target->notificationThreadList->end(), std::mem_fn(&std::thread::join));
                                               ^~
ViGEmClient/src/ViGEmClient.cpp:764:88: error: base operand of '->' is not a pointer
   std::for_each(target->notificationThreadList->begin(), target->notificationThreadList->end(), std::mem_fn(&std::thread::join));
                                                                                        ^~
ViGEmClient/src/ViGEmClient.cpp:764:115: error: 'std::thread' has not been declared
   std::for_each(target->notificationThreadList->begin(), target->notificationThreadList->end(), std::mem_fn(&std::thread::join));
                                                                                                                   ^~~~~~
ViGEmClient/src/ViGEmClient.cpp:765:34: error: request for member 'reset' in 'target->_VIGEM_TARGET_T::notificationThreadList', which is of non-class type 'int'
   target->notificationThreadList.reset();
                                  ^~~~~
ViGEmClient/src/ViGEmClient.cpp:766:36: error: cannot convert 'std::nullptr_t' to 'int' in assignment
   target->notificationThreadList = nullptr;
                                    ^~~~~~~
mingw32-make.exe: *** [Makefile:23: ViGEmClient.o] Error 1
